### PR TITLE
python3Packages.torchaudio-bin: init at 0.9.1

### PR DESCRIPTION
**Pull request by junjihashimoto**
###### Motivation for this change

As with torchvision-bin, this derivation is used with pytorch-bin to avoid the long build time of PyTorch CUDA builds.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/torchaudio/bin.nix [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  pyVerNoDot = builtins.replaceStrings [ "." ] [ "" ] python.pythonVersion;
+  srcs = import ./binary-hashes.nix version;
+  unsupported = throw "Unsupported system";
+  version = "0.9.1";
+in buildPythonPackage {
+  inherit version;
+
+  pname = "torchaudio";
+
+  format = "wheel";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/torchaudio/bin.nix:23]:
```suggestion
  pname = "torchaudio";
  inherit version;
  format = "wheel";
```

**Review by SuperSandro2000 — COMMENTED** [review]
Are there any other changes we did in pytorch-bin?

**Comment by junjihashimoto** [comment]
@SuperSandro2000 
This is the same as pytorch-bin's one.
https://github.com/NixOS/nixpkgs/pull/143525

**Comment by junjihashimoto** [comment]
But this is for pytorch-1.9.1. https://github.com/NixOS/nixpkgs/pull/143525 is for pytorch-1.10.0.

File: pkgs/development/python-modules/torchaudio/bin.nix [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+, pythonOlder
+, pythonAtLeast
+}:
+
+buildPythonPackage rec {
+  version = "0.9.1";
+
+  pname = "torchaudio";
+
+  format = "wheel";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/torchaudio/bin.nix:20]:
```suggestion
  pname = "torchaudio";
  version = "0.9.1";
  format = "wheel";
```

File: pkgs/development/python-modules/torchaudio/bin.nix [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+  src =
+    let pyVerNoDot = lib.replaceStrings [ "." ] [ "" ] python.pythonVersion;
+        unsupported = throw "Unsupported system";
+        srcs = (import ./binary-hashes.nix version)."${stdenv.system}-${pyVerNoDot}" or unsupported;
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/torchaudio/bin.nix:25]: I think stdenv.system got recently deprecated.

File: pkgs/development/python-modules/torchaudio/bin.nix [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    in fetchurl srcs;
+
+  disabled = ! (pythonAtLeast "3.7" && pythonOlder "3.10");
+
+  nativeBuildInputs = [
+    patchelf
+  ];
+
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/torchaudio/bin.nix:33]:
```suggestion
```
That should be part of the stdenv already.

File: pkgs/development/python-modules/torchaudio/bin.nix (hunk 3 above)
Comment (junjihashimoto) [reply pkgs/development/python-modules/torchaudio/bin.nix]: Is it top-level system variable? If so, it has changed.

Comment (junjihashimoto) [reply pkgs/development/python-modules/torchaudio/bin.nix:20]: CI throws an error with top-level system variable, so I've reverted it.

**Pull request closed without merging**